src/main/java/com/selfxdsd/storage/SelfInvoices.java

```diff
@@ -112,17 +124,29 @@ public Invoices ofContract(final Contract.Id id) {
      * @param record Record.
      * @return Invoice.
      */
-    private Invoice buildInvoice(final Record record){
+    private Invoice buildInvoice(final Record record) {
+        final Contract contract = this.storage.contracts().findById(
+            new Contract.Id(
+                record.getValue(SLF_INVOICES_XDSD.REPO_FULLNAME),
+                record.getValue(SLF_INVOICES_XDSD.USERNAME),
+                record.getValue(SLF_INVOICES_XDSD.PROVIDER),
+                record.getValue(SLF_INVOICES_XDSD.ROLE)
+            )
+        );
+        return this.buildInvoice(record, contract);
+    }
+
+    /**
+     * Builds an Invoice from a {@link Record} and a {@link Contract}.
+     *
+     * @param record Record.
+     * @param contract Contract.
+     * @return Invoice.
+     */
+    private Invoice buildInvoice(final Record record, final Contract contract) {
```

Comment:
@criske Why did you make two ``buildInvoice`` methods?

Reply:
I've wanted to cache the contract in `SelfInvoices#ofContract()`. By using the initial `buildInvoice`, that would require fetching the same contract from db as many times there is an invoice to be fetched.
For example if there are 10 invoices for  some `contract.id`, that would've result in calling `storage.contracts().findById` 10 times, instead of once.

Reply:
@rultor Right, makes sense. Thanks!